blscosi/protocol/sub_protocol.go

```diff
@@ -239,15 +248,16 @@ func (p *SubBlsCosi) dispatchSubLeader() error {
 
 	responses := make(ResponseMap)
 	for _, c := range p.Children() {
-		public := p.NodePublic(c.ServerIdentity)
+		_, index := searchPublicKey(p.TreeNodeInstance, c.ServerIdentity)
```

Comment:
If key is not found, index will be -1 and line 253 will panic. (Same for most other uses of searchPublicKey.)

Reply:
I should have verified but note that this is a map and then `-1` is a valid key. I changed into integers because I need an ordering for the coefficients but it is still a map to filter messages and take only the ones from known distant peers.

Reply:
Also if the public key is returned, the index will always be positive so other uses are safe with the nil check.